test/rust_src/src/alloc-tests.el

```diff
@@ -0,0 +1,12 @@
+;;; alloc-tests.el ---  -*- lexical-binding: t; -*-
+
+(require 'ert)
+
+(ert-deftest memory-use-count ()
+  (let ((counts (memory-use-counts)))
+    (should (= 8 (length counts)))
+    (dolist (count counts)
+      (should (integerp count)))))
+
+(provide 'alloc-tests)
```

Comment:
Is this necessary ?

Reply:
Do you mean the `provide` line? I don't know, `auto-insert` adds that and I always just leave it.

Reply:
It's only necessary if users want to do `(require 'alloc-tests)`. It's harmless and other Emacs test files do it (e.g. abbrev-tests.el).

If a user really wanted to define their own alloc-tests.el, it'd still be fine I think, because none of the tests (for C or Rust) are put on `load-path` for users.